src/AutoMapper/ExpressionExtensions.cs

```diff
@@ -11,6 +12,38 @@
 
     internal static class ExpressionExtensions
     {
+        public static Expression ForEach(Expression collection, ParameterExpression loopVar, Expression loopContent)
+        {
+            var elementType = loopVar.Type;
+            var enumerableType = typeof(IEnumerable<>).MakeGenericType(elementType);
+            var enumeratorType = typeof(IEnumerator<>).MakeGenericType(elementType);
+
+            var enumeratorVar = Variable(enumeratorType, "enumerator");
+            var getEnumeratorCall = Call(collection, enumerableType.GetMethod("GetEnumerator"));
```

Comment:
Yeah, I think that this was intending to emulate the for loop as close as possible. But it also doesn't dispose the enumerator, so who knows.